### PR TITLE
Added PDEX Token Handler

### PR DESCRIPTION
**Pull request by zktony**
Handled Native-token transfer from Solochain to Polkadex Parachain. 
Partially close https://github.com/Polkadex-Substrate/thea/issues/100

### CODE REVIEW

File: pallets/asset-handler/src/lib.rs [hunk 1]
```diff
@@ -660,6 +667,29 @@ pub mod pallet {
 			Ok(())
 		}
 
+		pub fn handle_asset(
+			asset_id: u128,
+			who: T::AccountId,
+			amount: u128,
+		) -> Result<(), DispatchError> {
```
Comment (serhii-temchenko) [inline pallets/asset-handler/src/lib.rs:680]: Please add documentation

File: pallets/asset-handler/src/lib.rs [hunk 2]
```diff
@@ -660,6 +667,29 @@ pub mod pallet {
...
+			let polkadex_asset_id = T::PolkadexAssetId::get();
+			if polkadex_asset_id == asset_id {
+				Self::lock_pdex_asset(amount, who)
+			} else {
+				Self::burn_thea_asset(asset_id, who, amount)
+			}
+		}
+
+		pub fn lock_pdex_asset(amount: u128, who: T::AccountId) -> DispatchResult {
```
Comment (serhii-temchenko) [inline pallets/asset-handler/src/lib.rs:694]: Same

File: pallets/thea/src/lib.rs [hunk 3]
```diff
@@ -503,13 +505,14 @@ pub mod pallet {
 		) -> Result<(), DispatchError> {
 			ensure!(beneficiary.len() <= 100, Error::<T>::BeneficiaryTooLong);
 			// TODO: This will be refactored when work on withdrawal so not fixing clippy suggestion
-			let (network, ..) = asset_handler::pallet::Pallet::<T>::get_thea_assets(asset_id);
+			let (mut network, ..) = asset_handler::pallet::Pallet::<T>::get_thea_assets(asset_id);
+			if asset_id == T::PolkadexAssetId::get() {
+				network = 1;
+			};
```
Comment (serhii-temchenko) [inline pallets/thea/src/lib.rs]: Please remove todo, looks like it's outdated and fixed already

Comment (serhii-temchenko) [inline pallets/thea/src/lib.rs]: With this approach extra query to the storage will be executed always, but it's can be optional and executed only if `asset_id != T::PolkadexAssetId::get()`. So I think it's could be changed to something like this:

```rust
			let network = if asset_id == T::PolkadexAssetId::get() {
				1
			} else {
				let (network, ..) = asset_handler::pallet::Pallet::<T>::get_thea_assets(asset_id);
				network
			};

```

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

**Review by serhii-temchenko — APPROVED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**